### PR TITLE
Update rbenv/bundler/github-pages instructions

### LINKED ISSUE #66: The local development instructions are tied to (ish) Mojave
The instructions for "previewing" the site locally have only been used successfully with Mojave: https://github.com/AlexsLemonade/training-specific-template#local-development

@cbethell ran into different permissions issues that the instructions didn't solve. If someone finds a solution, we should update that documentation.

### PR DESCRIPTION
**Pull request by jashapiro**
Since I was setting things up again for a new repo, I thought I could revisit and update the instructions for local development, going back through installation of rbenv. 

I'm doing this as a draft, because while it "works for me"™, that is no guarantee these days. So to give everyone a chance to look through and test it, so I am requesting everyone as a reviewer, but no rush.

Closes #66 (hopefully!)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 ### Workshops hosted by others
 
 If you are an external user, you can either [file an issue to request read access](https://github.com/AlexsLemonade/training-specific-template/issues/new?assignees=&labels=request&template=request-read-access-to-use-template-for-a-workshop.md&title=%5BRequest+read+access%5D+) or fork the repository.
-If you request read access and it is granted, follow [the instructions above](#ccdl-workshops).
+If you request read access and it is granted, follow [the instructions above](#data-lab-workshops).
 To fork the repository, [follow the GitHub instructions for forking a repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo).
 
 **There are currently some important limitations to using this repository as a template for externally-hosted workshop.**
-If you are not using the CCDL hosted RStudio Server, there are a number of instructions in this repository that are about distributing credentials and logging into that server.
+If you are not using the Data Lab hosted RStudio Server, there are a number of instructions in this repository that are about distributing credentials and logging into that server.
```
Comment (sjspielman) [inline README.md:24]: I feel like there should be some kind of dash between "Data Lab" and "hosted" but I will leave that grammar and type of dash to you!

File: README.md [hunk 2]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
-  brew update
-  brew install rbenv
+Installing the dependencies for GitHub pages is best done in a separate ruby environment, managed by [rbenv](https://github.com/rbenv/) and [Bundler](https://bundler.io)
 
-  # add to ~/.bash_profile
-  echo 'eval "$(rbenv init -)"' >> ~/.bash_profile
-  ```
+The following instructions are for installation on Mac OS.
+1. [Install rbenv](https://github.com/rbenv/rbenv#installation).
+The easiest, and recommended, installation is through Homebrew:
```
Comment (sjspielman) [inline README.md:70]: 👀 🍺

Comment (sjspielman) [reply README.md:70]: Do we want to include a "but first install homebrew" as needed? For Data Lab folks this seems implicit, but in case other teams use our materials to set up a workshop it seems more helpful for them.
Link - https://brew.sh/

File: README.md [hunk 3]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
+```
+Alternatively, you can install with [rbenv-installer](https://github.com/rbenv/rbenv-installer#rbenv-installer) which will work even without Homebrew (though it uses Homebrew if you have it).
 
-  Start a new shell to initialize `rbenv`.
+2. Initialize rbenv and install it for your shell:
+```
+ # run this and follow the printed instructions:
+rbenv init
+```
+Start a new shell to initialize `rbenv`.
```
Comment (sjspielman) [inline README.md:83]:
```suggestion
Then, start a new shell to initialize `rbenv`.
```

Comment (sjspielman) [reply README.md:83]: I'll note this only works for me if I add the `eval "$(rbenv init - zsh)"` line as recommended to my `.zshrc` _after_ the line that is popping homebrew paths into `PATH`, which makes sense but isn't explicitly spelled out.

File: README.md [hunk 4]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
 
-  Now you're ready to run `gem install bundler`. 
+4. Activate your ruby version. Here you have a few options:
 
-2. Install Jekyll (`gem install jekyll`).
-3. Run `bundle install` from the root of this repository, where the `Gemfile` is located.
-4. Run `bundle exec jekyll serve` and navigate to the server address.
+   a. If you want to set the global ruby version that will be used wherever you use `ruby`, you can use
+   ```
+   rbenv global 3.1.3
```
Comment (sjspielman) [inline README.md:95]: Noting I did this while testing

File: README.md [hunk 5]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
+   ```
 
-If you've already followed the steps above and are in a new shell, run the following steps to build locally:
+   b. To only use this version of ruby within the repository only, navigate to the root of this repository and use:
+   ```
+   rbenv local 3.1.3
+   ```
+   This will create a `.ruby-version` file that will automatically activate this version of ruby whenever you are in this directory.
 
+5. Now you should be ready to install Bundler:
```
Comment (sjspielman) [inline README.md:104]: Before this step you need to start a new shell for the `ruby` version to kick in, so the right `gem` executable is found. Otherwise my `gem` was `/usr/bin/gem`. In a new shell, `which gem` gave `/Users/sjspielman/.rbenv/shims/gem` which is the right one!

File: README.md [hunk 6]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
 ```
-rbenv shell 2.7.1
-rbenv rehash
-bundle exec jekyll serve  
+gem install bundler
 ```
+
+6. Finally, run `bundle install` from the root of this repository, where the `Gemfile` is located.
```
Comment (sjspielman) [inline README.md:109]: Again they would have needed a new shell to ensure the right `bundle` executable.

File: README.md [hunk 7]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
+This will install all additional dependencies.
+
+### Running a local jekyll server
+
+Once all dependencies are installed, you should be able to start a local jekyll server with:
+
+```
+bundle exec jekyll serve
```
Comment (sjspielman) [inline README.md:117]: At this point my `which jekyll` is still heading to an older `/usr/local/bin/jekyll`. Before we get here, I think it's worth adding a step `rbenv rehash` which will clean up all the executables to default to the `rbenv` ones. ("fixed" my path, `which jekyll` --> `/Users/sjspielman/.rbenv/shims/jekyll`).

But either way I'm getting a different build error: 
```
/Users/sjspielman/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/jekyll-3.9.2/lib/jekyll/commands/serve/servlet.rb:3:in `require': cannot load such file -- webrick (LoadError)
```

I looked into it and it seems that `webrick` no longer comes with ruby >=3 but has to be installed as a gem. I ran `bundle add webrick` to update the gemfile, and then it served without error!

In conclusion, **add `webrick` to the Gemfile**. Not sure of version though tbh...

**Review by sjspielman — COMMENTED** [review]
Mostly works for me, and found some fixes for areas it didn't!

File: README.md (hunk 5 above)
Comment (jashapiro) [reply README.md:104]: I would expect this to happen after the first shell restart. After the `rbenv init` instructions are followed (the `eval` line added to PATH), the changes should be instant.

File: README.md (hunk 7 above)
Comment (jashapiro) [reply README.md:117]: I did add webrick!

The version of jekyll you get with `which jekyll` does not matter if you use `bundle exec jekyll serve`. The point of that command is to use the bundled jekyll regardless of what else might be installed.

Comment (sjspielman) [reply README.md:117]: Ah womp, sorry, wasn't in the right branch. All set :)

File: README.md [hunk 8]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
+   rbenv local 3.1.3
+   ```
+   This will create a `.ruby-version` file that will automatically activate this version of ruby whenever you are in this directory.
 
+7. Now you should be ready to install Bundler:
 ```
-rbenv shell 2.7.1
-rbenv rehash
-bundle exec jekyll serve  
+gem install bundler
```
Comment (allyhawkins) [inline README.md:106]: Just noting that at first I was getting an error here: 
```
you don't have write permissions for the /library/ruby/gems/2.6.0 directory
```
I restarted my terminal and then ran the command again and no error, so maybe we need to add a line about starting a new shell? 
But other than that everything worked for me.

Comment (jashapiro) [reply README.md:106]: Did you start a new shell at step 4 (after modifying your `.bashrc` or `.zshrc`?

Comment (allyhawkins) [reply README.md:106]: Yes I did it after that step and then again when I saw that error.

Comment (jashapiro) [reply README.md:106]: So interesting! I would not expect it to require two restarts, but I guess maybe it does.

File: README.md [hunk 9]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
+The easiest, and recommended, installation is through Homebrew:
+```
+# install rbenv
+brew update
+brew install rbenv ruby-build
+```
+Alternatively, you can install with [rbenv-installer](https://github.com/rbenv/rbenv-installer#rbenv-installer) which will work even without Homebrew (though it uses Homebrew if you have it).
 
-  Start a new shell to initialize `rbenv`.
+2. Install rbenv for your shell for your shell.
```
Comment (jashapiro) [inline README.md:78]:
```suggestion
2. Install rbenv for your shell.
```

File: README.md (hunk 8 above)
Comment (jashapiro) [reply README.md:106]: Did you do a global or local activation? I wonder if it doesn't rehash for a global until restart...

Comment (allyhawkins) [reply README.md:106]: I did local first and then when it didn't work I tried global and still got the same error.

Comment (jaclyn-taroni) [reply README.md:106]: I just did a local activation. A restart was not required.

File: README.md [hunk 10]
```diff
@@ -54,46 +54,66 @@ We recommend that you link to `<url for repository's GitHub pages>/workshop/HOME
...
-  ```
-  # install rbenv
-  brew update
-  brew install rbenv
+Installing the dependencies for GitHub pages is best done in a separate ruby environment, managed by [rbenv](https://github.com/rbenv/) and [Bundler](https://bundler.io)
 
-  # add to ~/.bash_profile
-  echo 'eval "$(rbenv init -)"' >> ~/.bash_profile
-  ```
+The following instructions are for installation on Mac OS.
```
Comment (cbethell) [inline README.md:68]: Are there alternate instructions if not on a Mac OS? This line gives me the impression that I can scroll down to find alternate instructions or to find a link to alternate instructions.

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻 I had no issues (but I have also recently installed and initialized `rbenv` 😄 )

**Comment by jaclyn-taroni** [comment]
Oh, you might consider adding `.ruby-version` to `.gitignore` while you're at it?

File: README.md (hunk 8 above)
Comment (cbethell) [reply README.md:106]: I also received the same error as Ally here after doing a global activation -- which was also fixed by restarting the terminal.

File: README.md (hunk 7 above)
Comment (cbethell) [reply README.md:117]: I also received the same original error noted above in https://github.com/AlexsLemonade/training-specific-template/pull/101/files#r1101582245 and was able to get things working by running `bundle add webrick` as Stephanie mentioned then re-running this step!

**Review by cbethell — COMMENTED** [review]
I got everything here working with a few tweaks that I mention below!

File: README.md (hunk 7 above)
Comment (jashapiro) [reply README.md:118]: Just to confirm, you hadn't pulled this branch for testing, right? Because I did add webrick to the gemfile as part of this PR: https://github.com/AlexsLemonade/training-specific-template/blob/b3ca78851811ab3f08df78d7d41234066b7019d0/Gemfile#L4

**Review by sjspielman — APPROVED** [review]
All my remaining comments are at your discretion, so 🚀

File: README.md (hunk 2 above)
Comment (jashapiro) [reply README.md:71]:
```suggestion
The easiest, and recommended, installation is through [Homebrew](https://brew.sh/):
```
